compliance/index/v4/node.go

```diff
@@ -0,0 +1,182 @@
+	"github.com/stackrox/rox/pkg/scannerv4/mappers"
+)
+
+// The layer carries a hardcoded digest, as it is exclusively used for passing
+// ClairCore checks, not for Scanners matching
+var layerDigest = fmt.Sprintf("sha256:%s", strings.Repeat("a", 64))
+
+type localNodeIndexer struct {
+}
+
+// NewNodeIndexer creates a new node indexer
+func NewNodeIndexer() compliance.NodeIndexer {
+	return &localNodeIndexer{}
+}
+
+// GetIntervals
+func (l *localNodeIndexer) GetIntervals() *intervals.NodeScanIntervals {
+	i := intervals.NewNodeScanIntervalFromEnv()
+	return &i
+}
+
+// IndexNode indexes a live fs.FS at the container mountpoint given in the basePath.
+func (l *localNodeIndexer) IndexNode(ctx context.Context) (r *v4.IndexReport, err error) {
+	report := &claircore.IndexReport{
+		Packages:      map[string]*claircore.Package{},
+		Environments:  map[string][]*claircore.Environment{},
+		Distributions: map[string]*claircore.Distribution{},
+		Repositories:  map[string]*claircore.Repository{},
+		Files:         map[string]claircore.File{},
+	}
+
+	layer, err := constructLayer(ctx, layerDigest, env.NodeScanningV4HostPath.Setting())
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if tmpErr := layer.Close(); tmpErr != nil {
+			err = tmpErr
+		}
+	}()
+
+	reps, err := runRepositoryScanner(ctx, layer)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to run repository scanner")
+	}
+
+	pcks, err := runPackageScanner(ctx, layer)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to run package scanner")
+	}
+
+	ir, err := coalesceReport(ctx, layerDigest, reps, pcks)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to coalesce report")
+	}
+
+	report = controller.MergeSR(report, []*claircore.IndexReport{ir})
+	report.Success = true
+	report.State = controller.IndexFinished.String()
+
+	r, err = mappers.ToProtoV4IndexReport(report)
+	if err != nil {
+		return nil, errors.Wrap(err, "converting clair report to v4 report")
+	}
+
+	return
+}
+
+func coalesceReport(ctx context.Context, layerDigest string, reps []*claircore.Repository, pcks []*claircore.Package) (*claircore.IndexReport, error) {
+	ch, err := claircore.ParseDigest(layerDigest)
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing Clair Core digest")
+	}
+
+	la := &ccindexer.LayerArtifacts{
+		Hash: ch,
+	}
+	la.Repos = append(la.Repos, reps...)
+	la.Pkgs = append(la.Pkgs, pcks...)
+	artifacts := []*ccindexer.LayerArtifacts{la}
+	coal := new(rhel.Coalescer)
+
+	ir, err := coal.Coalesce(ctx, artifacts)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to coalesce report")
+	}
+	return ir, nil
+}
+
+func runPackageScanner(ctx context.Context, layer *claircore.Layer) ([]*claircore.Package, error) {
+	rpm := rpm2.Scanner{}
+	pck, err := rpm.Scan(ctx, layer)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to invoke RPM scanner")
+	}
+	pck = filterPackages(ctx, pck)
+	if pck != nil {
+		zlog.Info(ctx).Msgf("Num packages found: %v", len(pck))
```

Comment:
do we want to keep using the `zlog` thing here? This is now our compliance code so we could maybe use the same logger as we use in the rest of this code base?

Reply:
That's a great catch! Moved this over to our own logger. 
Though the ClairCore parts we're calling still use zlog, so we cannot get rid of it entirely in our console logs :)